sdc/datatypes/hpat_pandas_dataframe_functions.py

```diff
@@ -1516,3 +1516,109 @@ def sdc_pandas_dataframe_groupby_impl(self, by=None, axis=0, level=None, as_inde
         return init_dataframe_groupby(self, column_id, grouped, sort)
 
     return sdc_pandas_dataframe_groupby_impl
+
+
+def sdc_pandas_dataframe_reset_index_codegen(df_type, all_params, columns, result_name, func_line):
+    joined = ', '.join(all_params)
+    func_lines = [f'def _df_reset_index_impl({joined}):']
+    df = all_params[0]
+    func_lines += func_line
+    for i, c in enumerate(columns):
+        result_c = f'result_{c}'
+        func_lines += [
+            f'  series_{c} = pandas.Series(get_dataframe_data({df}, {i}))',
+            f'  result_{c} = series_{c}._data'
+        ]
+        result_name.append((result_c, c))
+    data = ', '.join(f'"{column_name}": {column}' for column, column_name in result_name)
+    func_lines += [f'  return pandas.DataFrame({{{data}}}, index=numpy.arange(len(result_{c})))']
+    func_text = '\n'.join(func_lines)
+
+    global_vars = {'pandas': pandas,
+                   'numpy': numpy,
+                   'get_dataframe_data': get_dataframe_data}
+
+    return func_text, global_vars
+
+
+def sdc_pandas_dataframe_reset_index_drop_False(df_type, result_name, func_lines):
+    codegen_ind = df_index_codelines(df_type, True)
+    result_ind = f'res_index'
+    for i in codegen_ind:
+        func_lines += [f'{i}']
+    result_name.append((result_ind, 'index'))
+
+    return result_name, func_lines
+
+
+def sdc_pandas_dataframe_reset_index_impl(self, level=None, drop=False, inplace=False, col_level=0, col_fill=''):
+    all_params = ['self', 'level=None', 'drop=False', 'inplace=False', 'col_level=0', 'col_fill=""']
+    df_func_name = f'_df_reset_index_impl'
+
+    result_name = []
+    func_lines = []
+    drop_val = str(drop)[14:-1]
+    if drop_val == 'False':
```

Comment:
Looks suspicious. Why do you compare with string?

Reply:
because here drop == Literal[bool](False) or Literal[bool](True)

Reply:
So, why not to use `drop.literal_value`?